search-commons/src/test/java/no/unit/nva/search/resource/UserSettingsClientTest.java

```diff
@@ -1,89 +1,134 @@
+    var contributorId = randomString();
+    var value =
+        FakeHttpResponse.create(
+            """
+            {
+              "promotedPublications": [
+                "https://api.dev.nva.aws.unit.no/publication/123",
+                "https://api.dev.nva.aws.unit.no/publication/456"
+              ]
+            }
+            """,
+            HTTP_OK);
+    when(httpClient.send(any(HttpRequest.class), eq(BodyHandlers.ofString()))).thenReturn(value);
+    var promotedPublications = userSettingsClient.fetchPromotedPublications(contributorId);
+
+    assertFalse(promotedPublications.isEmpty());
+  }
+
+  @Test
+  void shouldReturnEmptyListOnFailureAndLogResponseThatFailed()
+      throws IOException, InterruptedException {
+    var contributorId = randomString();
+    when(httpClient.send(any(HttpRequest.class), eq(BodyHandlers.ofString())))
+        .thenReturn(FakeHttpResponse.create("{}", HTTP_BAD_GATEWAY));
+
+    var logger = LogUtils.getTestingAppenderForRootLogger();
+    var promotedPublications = userSettingsClient.fetchPromotedPublications(contributorId);
+
+    assertTrue(
+        logger
+            .getMessages()
+            .contains(
+                String.format("Failed to fetch user settings for contributor %s", contributorId)));
+    assertTrue(promotedPublications.isEmpty());
   }
 
-  @ParameterizedTest
-  @MethodSource("uriProvider")
-  void searchWithUriReturnsOpenSearchAwsResponse(URI uri) throws ApiGatewayException {
-    var resourceAwsQuery =
-        ResourceSearchQuery.builder()
-            .fromTestQueryParameters(queryToMapEntries(uri))
-            .withRequiredParameters(FROM, SIZE)
-            .build();
-    var promotedPublications =
-        attempt(() -> userSettingsClient.doSearch(resourceAwsQuery, Words.RESOURCES))
-            .map(UserSettings::promotedPublications)
-            .orElse(logExceptionAndContinue());
-    assertNotNull(promotedPublications);
+  @Test
+  void shouldNotSendAuthorizationHeaderWhenRequestingPromotedPublications()
+      throws IOException, InterruptedException {
+    var contributorId = randomString();
+    when(httpClient.send(any(HttpRequest.class), eq(BodyHandlers.ofString())))
+        .thenReturn(FakeHttpResponse.create("{}", HTTP_BAD_GATEWAY));
+
+    userSettingsClient.fetchPromotedPublications(contributorId);
+
+    var requestCaptor = ArgumentCaptor.forClass(HttpRequest.class);
+    verify(httpClient).send(requestCaptor.capture(), eq(BodyHandlers.ofString()));
+    assertFalse(requestCaptor.getValue().headers().map().containsKey("Authorization"));
   }
 
-  private FunctionWithException<Failure<List<String>>, List<String>, RuntimeException>
-      logExceptionAndContinue() {
-    return (e) -> {
-      if (e.isFailure()) {
-        logger.error(e.getException().getMessage());
-      }
-      return List.of();
-    };
+  private record FakeHttpResponse(String body, int statuCode) implements HttpResponse<String> {
```

Comment:
I think my preference is to mock this instead of implementing fake

```java
    var response = mock(HttpResponse.class);
    when(response.statusCode()).thenReturn(HTTP_BAD_GATEWAY);
    when(response.body()).thenReturn("{}");
```

Reply:
I have tried, but today mockito does not like when I do mock inside of another mock:
```
Unfinished stubbing detected here:
-> at no.unit.nva.search.resource.UserSettingsClientTest.shouldReturnEmptyListOnFailureAndLogResponseThatFailed(UserSettingsClientTest.java:59)

E.g. thenReturn() may be missing.
Examples of correct stubbing:
    when(mock.isOk()).thenReturn(true);
    when(mock.isOk()).thenThrow(exception);
    doThrow(exception).when(mock).someVoidMethod();
```
I think fake response is ok + no need to suppress unchecked cast exception :)